spec/sorcery_crypto_providers_spec.rb

```diff
@@ -207,6 +207,36 @@
         expect(Sorcery::CryptoProviders::BCrypt.matches?(@digest, *@tokens)).to be true
       end
 
+      it 'matches? returns false when pepper is replaced with empty string' do
+        Sorcery::CryptoProviders::BCrypt.pepper = ''
+        expect(Sorcery::CryptoProviders::BCrypt.matches?(@digest, *@tokens)).to be false
+      end
+
+      it 'matches? returns false when no match' do
+        expect(Sorcery::CryptoProviders::BCrypt.matches?(@digest, 'a_random_incorrect_password')).to be false
+      end
+    end
+
+    context "when pepper is an empty string (default)" do
+      before(:each) do
+        Sorcery::CryptoProviders::BCrypt.pepper = ''
+        @digest = Sorcery::CryptoProviders::BCrypt.encrypt(@tokens) # a BCrypt::Password object
+      end
+
+      # make sure the default pepper '' does nothing
+      it 'matches token encrypted with salt only (without pepper)' do
+        expect(@digest).to eq @tokens.flatten.join # keep consistency with the older versions of #join_token
```

Comment:
the right hand side is actually current `#join_token` (versions without pepper).
This case is here to check the new token generated with default pepper (i.e. empty string) stays the same as the token that would have been generated by the current Sorcery.